src/full/Agda/Interaction/Highlighting/JSON.hs

```diff
@@ -0,0 +1,117 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+-- | Functions which give precise syntax highlighting info in JSON format.
+
+module Agda.Interaction.Highlighting.JSON (jsonifyHighlightingInfo) where
+
+import Agda.Interaction.Highlighting.Precise hiding (String)
+import Agda.Interaction.Highlighting.Range
+import Agda.Interaction.EmacsCommand
+import Agda.Interaction.Response
+import Agda.Syntax.Common
+import Agda.TypeChecking.Monad
+  (TCM, envHighlightingMethod, HighlightingMethod(..), ModuleToSource)
+import Agda.Utils.FileName
+import qualified Agda.Utils.IO.UTF8 as UTF8
+import Agda.Utils.String
+
+import qualified Control.Exception as E
+import Control.Monad.Reader
+import Data.Aeson
+import Data.ByteString.Lazy (ByteString)
+import qualified Data.ByteString.Lazy.Char8 as BS
+import Data.Char
+import qualified Data.Map as Map
+import Data.Maybe
+import Data.Monoid
+import qualified System.Directory as D
+import qualified System.IO as IO
+
+#include "undefined.h"
+import Agda.Utils.Impossible
+
+------------------------------------------------------------------------
+-- Read/show functions
+
+-- | Converts the 'aspect' and 'otherAspects' fields to atoms
+toAtoms :: Aspects -> [String]
```

Comment:
This is an exact copy of `toAtomes` from `Agda.Interaction.Highlighting.Emacs`.
You should create `Agda.Interaction.Highlighting.Common` and put such shared
definitions there to avoid code duplication.

Reply:
Will do!